minio_storage/files.py

```
@@ -0,0 +1,70 @@
+import tempfile
+from logging import getLogger
+
+from django.core.files.base import File
+from minio.error import ResponseError
+
+"""
+
+random test version based on stuff from here:
+
+https://github.com/jschneier/django-storages/blob/master/storages/backends/s3boto3.py#L68
+# noqa
+
+
+"""
+
+logger = getLogger("minio_storage")
+
+
+class MinioStorageFile(File):
+    def __init__(self, name, mode, storage,  **kwargs):
+
+        self._storage = storage
+        self.name = name
+        self._mode = mode
+        self.obj = storage.client.get_object(self._storage.bucket_name, name)
+        self._file = None
+
+
+class ReadOnlySpooledTemporaryFile(MinioStorageFile):
+    max_memory_size = 1024 * 1024 * 10
+
+    def __init__(self, name, mode, storage, max_memory_size=None, **kwargs):
+        if mode.find("w") > -1:
+            raise NotImplementedError(
+                "ReadOnlySpooledTemporaryFile storage cannot write to file")
+        if max_memory_size is not None:
+            self.max_memory_size = max_memory_size
+        super(ReadOnlySpooledTemporaryFile, self).__init__(name, mode, storage)
+
+    def _get_file(self):
+        if self._file is None:
+            try:
+                obj = self._storage.client.get_object(
+                    self._storage.bucket_name, self.name)
+                self._file = tempfile.SpooledTemporaryFile(
+                    max_size=self.max_memory_size)
+                for d in obj.stream(amt=1024 * 1024):
+                    self._file.write(d)
+                self._file.seek(0)
+                return self._file
+            except ResponseError as error:
```

thats a bit of separate issue from this one.. I mostly just moved some code over from the storage class..

I'm not sure which of those exception should be wrapped in other exceptions, the django docs doesnt hint very much at which exceptions and error are expected from storage classes.. 

NoSuchBucket does not seem like a very bad error to me,  if a configured storage bucket doesn't exist the application should probably fail fast instead of trying to manage the situation...  I guess the only difference between this a something like the regular File class is that it isn't a IOError which is raised anymore.. Not sure exactly how much of the detail.